homeassistant/components/solarlog/__init__.py

```diff
@@ -22,3 +27,46 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    if config_entry.version > 1:
+        # This means the user has downgraded from a future version
+        return False
+
+    if config_entry.version == 1:
+        if config_entry.minor_version < 2:
+            # migrate config_entry
+            new = {**config_entry.data}
+            new["extended_data"] = False
+
+            hass.config_entries.async_update_entry(
```

Comment:
You should update the config entry after the migration

Reply:
Done. What is the reasoning behind this?

Reply:
The migration will be only executed if the version doesn't match.

If an error occurs during the migration, the migration will never be executed again as the config entry has already been updated.